### PR TITLE
github: add make workflow

### PR DESCRIPTION
**Pull request by RiRa12621**
This removes the need to run make

### CODE REVIEW

File: .github/workflows/makecommit.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+jobs: 
+  run:
+    name: Make Generate Template
+    runs-on: ubuntu-latest
+    steps: 
+    - name: Checkout repo
+      uses: actions/checkout@v2
+
+    - name: Make
+      run: make clean && make
```
Comment (jewzaam) [inline .github/workflows/makecommit.yml:13]:
```suggestion
      run: make clean generate-syncset
```
This replicates the default target but it's explicit.

File: .github/workflows/makecommit.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    - name: Commit changes
+      uses: EndBug/add-and-commit@v4
+      with:
+        message: "hack: make generate"
+        add: "."
```
Comment (jewzaam) [inline .github/workflows/makecommit.yml:19]:
```suggestion
        add: "hack/00-osd-managed-cluster-config.selectorsyncset.yaml.tmpl"
```

File: .github/workflows/makecommit.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    - name: Checkout repo
+      uses: actions/checkout@v2
+
+    - name: Make
+      run: make clean && make
+
+    - name: Commit changes
+      uses: EndBug/add-and-commit@v4
+      with:
+        message: "hack: make generate"
```
Comment (jewzaam) [inline .github/workflows/makecommit.yml:18]:
```suggestion
        message: "on push: make generate-syncset"
```

Comment (RiRa12621) [reply .github/workflows/makecommit.yml:18]: is that our convention?
i'm always trying to structure things like "path/to/folder: what i changed"

**Comment by RiRa12621** [comment]
yes @jewzaam 
writing the action was quicker than i thought.
if i had known, i'd have put a hold on the hook PR

**Pull request merged**